app/main/views.py

```diff
@@ -120,10 +131,10 @@ def update(service_id, section):
         )
 
     if form.errors:
+        service.update(form.dirty_data)
```

Comment:
And re. `priceString` input names: I like how this means all of the validators are run in order and it's all hidden inside one field. 

But on the other hand this means that all of the Validate test methods are coupled to the `priceString` format (e.g. `max_price_not_a_number` and `min_price_not_a_number` do the same thing, so if the fields were separate we could just have `price_not_a_number`)

The way I've done "composite field" validation before is have steps that would run after individual field validators and have access to the full posted data, so that you could create and join fields like "priceString" there. I have no idea how hard it would be to do this in our case and if it would be better at all.